code/modules/power/geothermal/_geothermal.dm

```diff
@@ -0,0 +1,103 @@
+var/global/const/GEOTHERMAL_EFFICIENCY_MOD =   0.2
+var/global/const/GEOTHERMAL_PRESSURE_LOSS =    0.3
+var/global/const/MAX_GEOTHERMAL_PRESSURE =    2000
+var/global/const/GEOTHERMAL_PRESSURE_TO_POWER = 100
+
+/obj/machinery/power/geothermal
+	icon = 'icons/obj/machines/power/geothermal.dmi'
+	icon_state = "geothermal-base"
+	var/tmp/neighbors = 0
+	var/tmp/current_pressure = 0
+	var/tmp/processing = FALSE
+	var/efficiency = 0.5
+
+/obj/machinery/power/geothermal/RefreshParts()
+	..()
+	efficiency = Clamp(total_component_rating_of_type(/obj/item/stock_parts/capacitor) * GEOTHERMAL_EFFICIENCY_MOD, GEOTHERMAL_EFFICIENCY_MOD, 1)
+
+/obj/machinery/power/geothermal/proc/add_pressure(var/pressure)
+	current_pressure = Clamp(current_pressure + pressure, 0, MAX_GEOTHERMAL_PRESSURE)
+	if(!processing)
+		processing = TRUE
+		START_PROCESSING(SSmachines, src)
+
+/obj/machinery/power/geothermal/Process()
+	if(anchored && !(stat & BROKEN) && loc)
+		var/lost_pressure = current_pressure * 0.5
+		current_pressure -= lost_pressure
+		var/remaining_pressure = lost_pressure
+		lost_pressure = round(lost_pressure * efficiency)
+		remaining_pressure -= lost_pressure
+		if(powernet)
+			var/generated_power = round(lost_pressure * GEOTHERMAL_PRESSURE_TO_POWER)
+			if(generated_power)
+				add_avail(generated_power)
+		remaining_pressure = round(remaining_pressure * GEOTHERMAL_PRESSURE_LOSS)
+		if(remaining_pressure)
+			addtimer(CALLBACK(src, .proc/propagate_pressure, remaining_pressure), 5)
+	update_icon()
+	if(current_pressure <= 10)
+		processing = FALSE
+		return PROCESS_KILL
+
+/obj/machinery/power/geothermal/proc/propagate_pressure(var/remaining_pressure)
+	var/list/neighbors
+	for(var/neighbordir in global.cardinal)
+		var/obj/machinery/power/geothermal/neighbor = (locate() in get_step(loc, neighbordir))
+		if(neighbor?.anchored && !(neighbor.stat & BROKEN))
+			LAZYADD(neighbors, neighbor)
+	if(LAZYLEN(neighbors))
+		if(remaining_pressure)
+			for(var/obj/machinery/power/geothermal/neighbor AS_ANYTHING in neighbors)
+				neighbor.add_pressure(remaining_pressure)
```

Comment:
also we're giving full remaining_pressure to each, not spreading between them?
that'd make building more vents around central one plain better, which sounds okay from gameplay perspective, just kinda weird physics wise

Reply:
Should be divided by length of neighbors, not sure where that code went.